### PR TITLE
Adds the T-26 smart machinegun, replacing the old SG for smartgunners.

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Replaces the smartgun from SGs for the T-26 (because I think it's better.) This aims to improve the SG by a bit by ditching the powerpack system but reducing their longterm ammo unless they want to give up space for ammo, and no more free harness. Though I might re-add that.
SG is still on FC, not sure if I'll replace it later or not.
M56 NVGs no longer require the powerpack.
I'VE GOTTEN SPROTES
## Why It's Good For The Game
I feel like SG is a left behind role. I hope this improves that.

## Changelog
:cl:
add: T-26 Smart machinegun added. Added a new pocket for the M56 and renamed it to M26, it can carry the ammo drums.
del: SGs can no longer get m56bs
tweak: You no longer require a powerpack to put on SG NVGs.
balance: SGs get the shiny new T-26 smart machinegun.
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -2. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by Pariah919** [comment]
I'd like a DNM (but not a do not tm) until I get real mob sprites please

**Comment by spookydonut** [comment]
is this just a new sprite and rename?  why not just call it something generic like "smart machinegun" instead

**Comment by jroinc** [comment]
-1

Ups ammo storage massively, more-or-less halved wield delay, more-or-less halved movedelay, adds free night-vision goggles with no downside or equipment to use.

This is just taking the smartgun and massively powercreeping it from it's current role (minimal damage that can fire through marines) into it's old role (4-5 of these fucks sprinting into the darkness full-autoing anything).

**Comment by Pariah919** [comment]
> Ups ammo storage massively, more-or-less halved wield delay, more-or-less halved movedelay, adds free night-vision goggles with no downside or equipment to use.
Autoloaders give you more effective ammo anyway than the old SG (300 vs 500) Though I might make it 250 instead. Move delay went from 1.6 to 0.9. I personally believe that 1.6 is way too much with stamina existing, it's still higher than every pfc gun. The highest being 0.75. It never got adjusted for stamina.
>This is just taking the smartgun and massively powercreeping it from it's current role (minimal damage that can fire through marines) into it's old role (4-5 of these fucks sprinting into the darkness full-autoing anything).
It deals far less effective damage than the old SG, as it has no burst fire with 0.1 fire delay. This is 4 rounds a second rather than 2 bursts a second. The DPS is literally 80, it has worse damage per second than an assault rifle. It also doesn't autoclip unless you magnetic harness but then you get worse overall accuracy than a smartgun.

>EDIT- Upon thinking about this more, the only thing I think is a big mistake is lowering the movedelay. These people shouldn't be in the front.

It's higher than every other PFC gun in the game.

**Comment by Pariah919** [comment]
@psykzz i got sprites now

**Comment by MiniMeatwad** [comment]
no burst fire plz fix

**Comment by MiniMeatwad** [comment]
no mag harness or extended barrel plz fix

**Comment by MiniMeatwad** [comment]
ah yes, the baby gunner, shoots peas that go through marines, does no useful damage and is detrimental to the team, very good pr +1 plz add im xwno main

File: code/datums/supplypacks.dm [hunk 1]
```diff
@@ -1657,6 +1657,18 @@ AMMO
...
 	group = "Ammo"
 
+/datum/supply_packs/ammo_smartmachinegun
+	name = "T26 smartmachinegun ammo crate (x3)"
+	contains = list(
+                    /obj/item/ammo_magazine/standard_smartmachinegun,
+                    /obj/item/ammo_magazine/standard_smartmachinegun,
+					/obj/item/ammo_magazine/standard_smartmachinegun
+					)
+	cost = 30
```
Comment (Rohesie) [inline code/datums/supplypacks.dm:1667]: Are you using tabs or spaces here?

File: code/modules/projectiles/magazines/rifles.dm [hunk 2]
```diff
@@ -219,6 +219,21 @@
...
+/obj/item/ammo_magazine/standard_smartmachinegun
+	name = "\improper T-29 drum magazine (10x24mm)"
+	desc = "A 10mm drum magazine."
+	caliber = "10x26mm caseless"
+	icon_state = "t29"
+	w_class = WEIGHT_CLASS_NORMAL
+	default_ammo = /datum/ammo/bullet/smartgun
+	max_rounds = 300
+	gun_type = /obj/item/weapon/gun/rifle/standard_smartmachinegun
+	reload_delay = 20
```
Comment (Rohesie) [inline code/modules/projectiles/magazines/rifles.dm:234]:
```suggestion
	reload_delay = 2 SECONDS
```

File: code/datums/supplypacks.dm (hunk 1 above)
Comment (Pariah919) [reply code/datums/supplypacks.dm]: should've fixed this I think. I did an oops.

File: code/modules/projectiles/gun_attachables.dm [hunk 3]
```diff
@@ -828,6 +835,20 @@ Defined in conflicts.dm of the #defines folder.
 	scatter_mod = 0
 	movement_acc_penalty_mod = 0
 
+/obj/item/attachable/stock/t29stock
+	name = "T-29 Stock"
```
Comment (Rohesie) [inline code/modules/projectiles/gun_attachables.dm:839]:
```suggestion
	name = "T-29 stock"
```

File: code/modules/projectiles/gun_attachables.dm [hunk 4]
```diff
@@ -828,6 +835,20 @@ Defined in conflicts.dm of the #defines folder.
...
+	desc = "A standard MG Stock."
```
Comment (Rohesie) [inline code/modules/projectiles/gun_attachables.dm:840]:
```suggestion
	desc = "A standard machinegun stock."
```

File: code/modules/clothing/suits/marine_armor.dm [hunk 5]
```diff
@@ -208,8 +208,8 @@
 	desc = "A well-crafted suit of M3 Pattern Armor typically found in the hands of higher-ranking officers. Useful for letting your men know who is in charge when taking to the field"
 
 /obj/item/clothing/suit/storage/marine/smartgunner
-	name = "M56 combat harness"
-	desc = "A heavy protective vest designed to be worn with the M56 Smartgun System. \nIt has specially designed straps and reinforcement to carry the Smartgun and accessories."
+	name = "M26 combat harness"
+	desc = "A heavy protective vest designed to be worn with the T26 Smartmachinegun System. \nIt has specially designed straps and reinforcement to carry the Smartmachineguns drums inside of it."
```
Comment (Rohesie) [inline code/modules/clothing/suits/marine_armor.dm:212]:
```suggestion
	desc = "A heavy protective vest designed to be worn with the T26 Smartmachinegun System. \nIt has specially designed straps and reinforcement to carry the smartmachinegun drums inside of it."
```

File: code/modules/clothing/glasses/night.dm [hunk 6]
```diff
@@ -42,8 +42,8 @@
 
 
 /obj/item/clothing/glasses/night/m56_goggles
-	name = "\improper M56 head mounted sight"
-	desc = "A headset and goggles system for the M56 Smartgun. Has a low-res short range imager, allowing for view of terrain."
+	name = "\improper M26 head mounted sight"
+	desc = "A headset and goggles system for use with the T-26 Smart machinegun. Has a low-res short range imager, allowing for view of terrain."
```
Comment (Rohesie) [inline code/modules/clothing/glasses/night.dm:46]:
```suggestion
	desc = "A headset and goggles system for use with the T-26 smart machinegun. Has a low-res short range imager, allowing for view of terrain."
```

File: code/modules/projectiles/guns/rifles.dm [hunk 7]
```diff
@@ -629,6 +629,48 @@
 	fire_delay = 1.1 SECONDS
 	burst_amount = 1
 
+//-------------------------------------------------------
+//T-29 Smart Machine Gun (It's more of a rifle than the SG.)
+
+/obj/item/weapon/gun/rifle/standard_smartmachinegun
+	name = "\improper T-29 smart machine gun"
+	desc = "The T-29 LMG is the TGMC's current standard IFF-capable Machine gun. It's known for its ability to lay down heavy fire support very well. It is generally used when someone wants to hold a position or provide fire support. Requires special training and it cannot turn off IFF. It uses 10x26mm ammunition."
```
Comment (Rohesie) [inline code/modules/projectiles/guns/rifles.dm:637]:
```suggestion
	desc = "The T-29 LMG is the TGMC's current standard IFF-capable light machine gun. It's known for its ability to lay down heavy fire support very well. It is generally used when someone wants to hold a position or provide fire support. Requires special training and it cannot turn off IFF. It uses 10x26mm ammunition."
```

Comment (Pariah919) [reply code/modules/projectiles/guns/rifles.dm:637]: >light machinegun
>10x26mm
💢

**Comment by Pariah919** [comment]
Unless I get more feedback or bugs to squash I'm personally not touching this anymore, and feel that it's ready to be merged.

**Comment by Max-023** [comment]
You forgot about the FC. They should get the SG too. I would suggest giving the FC the officer armor just so they have something to replace the harness.

**Comment by Pariah919** [comment]
I didn't give it to the FC because I wanted a limit of 4 per round. More than that and it starts to be too much as it's currently made.
FC needs a kit overhaul really.

File: code/modules/clothing/suits/marine_armor.dm [hunk 8]
```diff
@@ -223,7 +223,19 @@
 					/obj/item/explosive/mine,
 					/obj/item/weapon/combat_knife,
 					/obj/item/weapon/gun/smartgun,
+                    /obj/item/weapon/gun/rifle/standard_smartmachinegun,
```
Comment (psykzz) [inline code/modules/clothing/suits/marine_armor.dm:226]: indent

Comment (Pariah919) [reply code/modules/clothing/suits/marine_armor.dm:226]: aaaaaaaaaaaaaaaaa

**Comment by Hathkar** [comment]
Some minor feedback: I feel the sound effect is a bit too loud.
Also I miss the old smartgun noise.

**Comment by Mouseofthecake** [comment]
Marines have caught on and are maxing out smartgunner slots. There is literally no counter to five SGs shooting hundreds of rounds at you with impunity.

**Comment by Pariah919** [comment]
@Mouseofthecake it has the old sunder currently. It should have the same fire rate as an assault rifle. I'll make it be 0.225 fire delay

**Comment by Rohesie** [comment]
Code looks fine.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-4). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**